x-pack/plugins/watcher/public/application/models/watch/base_watch.js

```diff
@@ -79,9 +79,12 @@ export class BaseWatch {
   };
 
   get displayName() {
+    const saveName = this.name;
+
     if (this.isNew) {
       return i18n.translate('xpack.watcher.models.baseWatch.displayName', {
-        defaultMessage: 'New Watch',
+        defaultMessage: '{saveName}',
```

Comment:
i18n isn't needed here since the name does not need to be translated. Also, the name is an optional field, so there is a possibility that it will be undefined. As a result, this may result in:

<img width="439" alt="Screen Shot 2020-07-28 at 9 31 24 PM" src="https://user-images.githubusercontent.com/5226211/88746387-6dd12780-d11a-11ea-9da6-04f2e9fa985a.png">

I think the code could be simplified to:
```
  if (this.name) {
      return this.name;
    } else {
      return this.id;
    }
```

Reply:
@alisonelizabeth thanks you for reviewing my fix and providing such helpful advise. I'm implemented the change and hopefully the build will go well.